pkg/sql/opt/norm/join_funcs.go

```diff
@@ -551,9 +551,13 @@ func (c *CustomFuncs) ExtractJoinEquality(
 		newFilters,
 		private,
 	)
+	outputCols := leftCols
+	if joinOp != opt.SemiJoinOp && joinOp != opt.AntiJoinOp {
```

Comment:
[nit] maybe explain here that semi and anti joins don't output columns from the right side, unless you think that's obvious enough that an explanation isn't warranted.